examples/scenarioConstrainedDynamics.py

```diff
@@ -0,0 +1,277 @@
+
+The script is found in the folder ``basilisk/examples`` and executed by using::
+
+    python3 scenarioConstrainedDynamics.py
+
+The scenario outputs two plots: one for the translational constraint violations, and another for the rotational constraint 
+violations. These constraint violations can be used to evaluate whether the simulation is replicating coupled motion with
+the desired fidelity and tune gains if the bounds should be changed. Increasing alpha and beta will reduce the magnitude 
+of the constraint violations while increasing run time.
+
+Illustration of Simulation Results
+----------------------------------
+
+::
+
+    show_plots = True, env = 'Gravity'
+
+Here, two identical spacecraft are connected along their body frame x-axes with a slight tumble rate while in low Earth orbit. 
+The time history for both the holonomic direction and attitude constraint violations are shown below. Note the y-axis log scale 
+showing the order of magnitude of the violations. The initial transient rises from perfect initial setup conditions before 
+settling to an equilibrium. This equilibrium is impacted by how dynamic the scenario is (rotation rate, gravity, thrusting), 
+the mass and inertia of the respective vehicles, and how stiff the gains are driven by the alpha and beta tuning parameters.
+
+.. image:: /_images/Scenarios/scenarioConstrainedDynamicsDirectionConstraint.svg
+   :align: center
+
+.. image:: /_images/Scenarios/scenarioConstrainedDynamicsAttitudeConstraint.svg
+   :align: center
+
+"""
+
+#
+#   Basilisk Scenario Script
+#
+#   Purpose:            Illustrates two spacecraft connected via a holonomic constraint
+#   Author:             Andrew Morell
+#   Creation Date:      Jul 20, 2023
+#
+
+# system imports
+import os
+
+# Basilisk imports
+from Basilisk.architecture import messaging
+from Basilisk.utilities import (SimulationBaseClass, orbitalMotion, macros, RigidBodyKinematics)
+from Basilisk.simulation import (spacecraft, constraintDynamicEffector, gravityEffector, svIntegrators)
+import matplotlib.pyplot as plt
+
+# Utility imports
+import numpy as np
+
+from Basilisk import __path__
+bskPath = __path__[0]
+fileName = os.path.basename(os.path.splitext(__file__)[0])
+
+def run(show_plots, env):
+    """
+    The scenario can be run with the following setup parameters:
+
+    Args:
+        show_plots (bool): Determines if the script should display plots
+        env (str): Choose whether in Earth orbit 'Gravity' or deep space 'NoGravity'
+
+    """
+
+    # Create a sim module as an empty container
+    scSim = SimulationBaseClass.SimBaseClass()
+
+    # Create simulation variable names
+    simTaskName = "simTask"  # arbitrary name (don't change)
+    simProcessName = "simProcess"  # arbitrary name (don't change)
+
+    # Create the simulation process and specify the integration update time
+    simulationTimeStep = macros.sec2nano(0.01)  # update process rate update time
```

Comment:
To speed up the example script, let's change this to 0.1 seconds.  This still works well as an example script and cuts the simulation time in half.

Reply:
I agree that this doesn't lose any demonstration capabilities. I've increased the timestep to 0.1 seconds.